### PR TITLE
refactor: use functional keyboard building

### PR DESCRIPTION
**Pull request by KnorpelSenf**
PoV for https://github.com/grammyjs/grammY/issues/104. The relevant commit reduces keyboard building LoC to 68 % of its previous value.

### CODE REVIEW

File: source/index.ts [hunk 1]
```diff
@@ -72,28 +70,22 @@ async function endlessFailing(
...
+			const r = await context.wb.reader(category)
+			return [r.label(), `category:${category}`]
+		}))
+	const keyboard = InlineKeyboard.from(
+		data
+			.map(([label, cb]) => InlineKeyboard.text(label, cb))
+			.map(button => InlineKeyboard.row(button)),
 	)
-	const sorted = buttons
-		.sort((a, b) => a.text.localeCompare(b.text, context.wb.locale()))
```
Comment (EdJoPaTo) [inline source/index.ts:100]: isnt the sorting missing now?

File: source/index.ts [hunk 2]
```diff
@@ -28,23 +26,17 @@ if (process.env['NODE_ENV'] !== 'production') {
...
-bot.use(async (ctx, next) => {
-	try {
-		if (next) {
-			await next()
-		}
-	} catch (error: unknown) {
-		console.log('try send error', error)
-		await ctx.reply('😣 This happens… Please try again.')
-	}
+const safe = bot.errorBoundary(async ({error, ctx}) => {
```
Comment (EdJoPaTo) [inline source/index.ts:29]: I think I would prefer `const bot = new Bot<Context>(token).errorBoundary(…` more. having it named safe is somehow weird.

Also: neat that errorBoundary exists now, way more useful than `bot.catch` in my opinion.

File: source/index.ts (hunk 1 above)
Comment (KnorpelSenf) [reply source/index.ts:100]: Oh. Yes.

File: source/index.ts (hunk 2 above)
Comment (KnorpelSenf) [reply source/index.ts:29]: > neat that errorBoundary exists now

It was added in grammY 1.1 more than 2 years ago.

> I would prefer `const bot = new Bot<Context>(token).errorBoundary(…` more

This does not type-check.

Comment (EdJoPaTo) [reply source/index.ts:29]: Then rename the variable `safe` to `bot` and the current bot to something like `mainBot` or `botMain` or something like that? `safe` is not telling much about what is included and `bot` is the main thing to be worked with so it should be the short default. Or is there any other community default that evolved already?

File: package.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
 		"node": ">=18"
 	},
 	"dependencies": {
-		"grammy": "^1.10.0",
+		"grammy": "github:grammyjs/grammy#keyboard-from",
```
Comment (EdJoPaTo) [inline package.json:21]:
```suggestion
		"grammy": "^1.17.0",
```

**Review by EdJoPaTo — CHANGES_REQUESTED** [review]
@KnorpelSenf Would you like to get this implemented with the new grammY release?

**Comment by KnorpelSenf** [comment]
I mainly did this for testing purposes and to have something to discuss over. I am not interested in polishing this experiment. Please take over or close.

**Pull request closed without merging**